### DISCUSSION TITLE
Feature Request(s)

### DISCUSSION THREAD

**Discussion by jscottelblein** [JulianPrieber/littlelink-custom#138] [Feature requests]
I'm currently in the process of testing out LLC to see if it's a good replacement for my current LinkTree, and adding some links to it.

A few things I think would be nice additions to LLC to make things easier and more configurable.

- For button images, it'd be nice to have a filed to enter a URL for the image to use, as well and the ability to drop/upload an image. Currently I have 1 site that using the "custom website" option pulls the wrong image that I want to use and I cant see any other way to specify the exact image I want.
- It would also be great to allow setting the image alignment. I personally, like the look Linktree uses where it puts the image all the way to the left of the button:

![image](https://user-images.githubusercontent.com/1524526/186966553-be82eba6-9eab-44b1-baf3-3577fa067d63.png)

- Ability to choose whether to use your entered text, or the default word for the button. For example, the twitter button. I'd like to enter my @name as the text for the button, since people by now automatically know the Twitter color and that it uses the "@" format, but it would also be cool to have the Twitter bird icon along with my entered text.
- For certain site, like twitter, the user can just enter their @name instead of having to paste the whole URL, since that's the only difference between all user's (LinkTree does this).
- On the Profile page, a link that takes you to your profile right on it. (There's one in the admin/users section too, but it would be handy and intuitive if there were on on the actual profile page as well).
- Also on the profile page, the ability to add/upload your profile picture right there.

I know it's a bunch of stuff I'm bombing ya with, lol, but I realize it's a new project just getting off the ground, so thought the user feedback would be useful. =)

**Comment by JulianPrieber** [comment]
> - For button images, it'd be nice to have a filed to enter a URL for the image to use, as well and the ability to drop/upload an image. Currently I have 1 site that using the "custom website" option pulls the wrong image that I want to use and I cant see any other way to specify the exact image I want.

This has been asked for a couple of times. I never got very far into implementing this.
I would like to have a self-contained system that allows users to upload their own images for button icons.
I cannot really think of a system that implements this without using a database entry to remember the location of the image for each button.

Sadly, our updater currently does not support adding database entries, so it might take a while until I can work on this one properly.

<br>

> - It would also be great to allow setting the image alignment. I personally, like the look Linktree uses where it puts the image all the way to the left of the button

This is something I actually looked into and is definitely possible with themes.
I haven't managed to implement this into a theme properly. The way icons are implemented currently, it is not very easy to move ions around properly.
I'm not the best at working with HTML, so this one might be on me.

<br>

> - Ability to choose whether to use your entered text, or the default word for the button. For example, the twitter button. I'd like to enter my @name as the text for the button, since people by now automatically know the Twitter color and that it uses the "@" format, but it would also be cool to have the Twitter bird icon along with my entered text.

<br>

This one might not be so easy to implement, so I'm not sure if this will ever be added.
I will probably enable custom text on any button (like on the custom button). This might be a good compromise for now.

> - For certain site, like twitter, the user can just enter their @name instead of having to paste the whole URL, since that's the only difference between all user's (LinkTree does this).

Sounds good. Maybe I'll add this.

<br>

> - On the Profile page, a link that takes you to your profile right on it. (There's one in the admin/users section too, but it would be handy and intuitive if there were on on the actual profile page as well).

This might be added in the future, I'm not sure yet.

<br>

> - Also on the profile page, the ability to add/upload your profile picture right there.

Probably not happening unless I get some former education in the web dev field.

<br><br>

Thank you for the feedback @STaRDoGG!

**Reply by jscottelblein** [reply]
> This is something I actually looked into and is definitely possible with themes.
> I haven't managed to implement this into a theme properly. The way icons are implemented currently, it is not very easy to move ions around properly.
> I'm not the best at working with HTML, so this one might be on me.

In `brands.css` under the `.icon` selector you can add:

```
position: absolute;
left: 15px;
top: 15px;
```

And it'll look like this:

![image](https://user-images.githubusercontent.com/1524526/187599580-19259d7c-48e9-4f1d-b151-1128bcdfd400.png)

My 2 cents, I'm not really sure the `brands.css` is a good file to include in the themes template, since they're for brands it's likely they'll never really want to be changed, and any custom changes to it for a single version of a theme will always need to be re-added if any new brand gets added to the default file.

It might be good to include a link to a `custom.css` that gets loaded last of all other .css files and it can be part of the default theme template. Then anything a theme author adds to it will just overwrite that selector and never need to edit the brands file. It would also be a good place for the user to add any other random, smaller 'tweaks'.

Also, just a quick heads-up; I notice there's a broken image at `/studio/button-editor/1`

![image](https://user-images.githubusercontent.com/1524526/187603067-b01265cd-4844-48da-b7f4-f620c13b05a1.png)

it links to `./images/close.png`
